test/Test/Data/Array.purs

```diff
@@ -237,6 +253,14 @@ testArray = do
   assert $ A.modifyAtIndices [0, 2, 8] not [true,  true, true,  true] ==
                                            [false, true, false, true]
 
+  log "scanl should return an array that stores the accumulated value at each step"
+  assert $ A.scanl (+)  0 [1,2,3] == [1, 3, 6]
```

Comment:
Can we test that these agree with the Foldable versions please? It's probably not necessary to compare with the Foldable versions for the other functions, but this one differs from Haskell and so there's more than one "sensible" option for how to implement it, so I think it's worth being a little bit more careful/deliberate in the tests.

Reply:
Could you clarify what you mean by this? The tests I used were taken from the [`scanl` docs](https://pursuit.purescript.org/packages/purescript-foldable-traversable/4.1.1/docs/Data.Traversable#v:scanl) and [`scanr` docs](https://pursuit.purescript.org/packages/purescript-foldable-traversable/4.1.1/docs/Data.Traversable#v:scanr)

Reply:
Rereading this again, I think you mean this:
```purescript
assert $ A.scanl (+)  0 [1,2,3] == scanl (+) 0 [1,2,3]
```

Reply:
Yeah, exactly.